### DISCUSSION TITLE
412: {"error":"Account xxx must be registered in the current region

### DISCUSSION THREAD

**Discussion by jopocop** [llamafilm/tesla-http-proxy-addon#155] [Q&A]
I have exactly same issue that was reported in https://github.com/alandtse/tesla/issues/940, but forwarded to tesla proxy addon - but didn't find anything about this topic here.

Most of the installation steps done as described in https://github.com/alandtse/tesla/wiki/Proxy-setup-with-DuckDNS-and-NGinx, Tesla Custom Integration installed and started - but it fails due to:
```
DEBUG (MainThread) [teslajsonpy.connection] 412: {"error":"Account xxx must be registered in the current region https://fleet-api.prd.eu.vn.cloud.tesla.com, please see https://developer.tesla.com/docs/fleet-api/endpoints/partner-endpoints#register"}`
```
My location is Finland. Tesla account locale in Finland/Finnish. Car is in Finland (and I'm the first owner, MY23). In Tesla Proxy region is "Europe, Middle East, Africa", although I tried the others as well. In developer.tesla.com allowed origin is https://xxx.duckdns.org and redirect URI is same but with /callback. Pem-file can be found and Tesla Proxy seems running ok. In the log there is a:
```
2024-10-30T19:56:36+02:00 [info ] Received GET request for /api/1/products
2024-10-30T19:56:36+02:00 [debug] Forwarding request to https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/products
```
every time I try to start the Tesla Custom Integration.

I sort of feel that I'm on the final straight of a marathon, but getting cramps. Any hints?

**Comment by llamafilm** [comment]
It sounds like your public key is correctly reachable. I'm guessing this addon did not finish registering with your Tesla developer account. This might happen if you have been changing the domain while troubleshooting. So for example, the first time you ran this addon setup flow, it registered the Tesla account, and then something changed, and the add-on does not re-register because it thinks that's already done. I suggest turning on the "regenerate auth" setting and redoing the addon setup.

**Reply by jopocop** [reply]
In short: Thank you!

That was the final trick, the rest of the process was all sunshine. The 2-month installation journey is over, and the car is right now scheduled to start charging in night on cheapest hours using jonasbkarlsson's EV charging integration. (That will though change when wife comes home with her car). But integration works and is tested. 

My biggest problem throughout the process was probably that NGINX had to be started with a completely dummy domain at some point (before starting tesla proxy). I still don't understand the details (restarted nginx half a dozen of times even after this final hint). Meaning not either of the duckdns created domains, but just some gibberish, then start teslaproxy, and eventually restart nginx with real domain (which I suppose is actually not really needed by teslaproxy anyway). And, when I after 1.5 months finally got the public key visible to the whole world, I hesitated to mess up with the nginx config... Exactly what I should have done.

I hope nobody faces the same, but if you do, hope this q&a helps!

**Reply by Wintherman** [reply]
Hi all reading this.
Finally I also got it working after some weeks struggling based on jopocop text above.
Maybe it was the "regenerate auth" setting from llamafilm that made it.
Or it was the NGINX dummy domain when starting up the Tesla HTTP Proxy.
I started first the Tesla Proxy without my Tesla credentials.
I restarted NGINX and the Tesla Proxy several time and added the Tesla Credentials after some Restarts.
Not so structured I am afraid to say.
I had not much hope when I pressed Enroll public...". I had got so many rejections at that stage before.
So sorry to say I do not know extactly what did it.  But now I know much more about what NGINX is and how to set up logging in HA....
